dash.el

```diff
@@ -2275,7 +2275,13 @@ or with `-compare-fn' if that's non-nil.
 
 Alias: `-uniq'"
   (let (result)
-    (--each list (unless (-contains? result it) (!cons it result)))
+    (if (memq -compare-fn '(eq eql equal))
```

Comment:
`nil` is also a valid value here, and I suggest sorting the list of functions by popularity: `(nil equal eq eql)`.